utils/oscap-info.c

```diff
@@ -174,17 +261,263 @@ static inline void _print_xccdf_benchmark(struct xccdf_benchmark *bench, const c
+	struct oscap_string_iterator* checklist_it = ds_stream_index_get_checklists(stream);
+
+	printf("\nStream: %s\n", ds_stream_index_get_id(stream));
+	printf("Generated: %s\n", ds_stream_index_get_timestamp(stream));
+	printf("Version: %s\n", ds_stream_index_get_version(stream));
+
+	printf("Checklists:\n");
+	while (oscap_string_iterator_has_more(checklist_it)) {
+		const char * id = oscap_string_iterator_next(checklist_it);
+		printf("\tRef-Id: %s\n", id);
+
+		/* decompose */
+		struct oscap_source *xccdf_source = ds_sds_session_select_checklist(session, ds_stream_index_get_id(stream), id, NULL);
+		if (xccdf_source == NULL) {
+			oscap_string_iterator_free(checklist_it);
+			ds_stream_index_iterator_free(sds_it);
+			ds_sds_session_free(session);
+			return OSCAP_ERROR;
+		}
+
+		if (oscap_source_get_scap_type(xccdf_source) == OSCAP_DOCUMENT_XCCDF) {
+			struct xccdf_benchmark* bench = xccdf_benchmark_import_source(xccdf_source);
+			if(!bench) {
+				oscap_string_iterator_free(checklist_it);
+				ds_stream_index_iterator_free(sds_it);
+				ds_sds_session_free(session);
+				return OSCAP_ERROR;
+			}
+			_print_xccdf_benchmark(bench, prefix, 0);
+			// bench is freed as a side-effect of the function above
+		} else if (oscap_source_get_scap_type(xccdf_source) == OSCAP_DOCUMENT_XCCDF_TAILORING) {
+			_print_xccdf_tailoring(xccdf_source, prefix, 0);
+		}
+		ds_sds_session_reset(session);
+	}
+	oscap_string_iterator_free(checklist_it);
+
+	printf("Checks:\n");
+	struct oscap_string_iterator* checks_it = ds_stream_index_get_checks(stream);
+	while (oscap_string_iterator_has_more(checks_it)) {
+		const char * id = oscap_string_iterator_next(checks_it);
+		printf("\tRef-Id: %s\n", id);
+	}
+	oscap_string_iterator_free(checks_it);
+
+	struct oscap_string_iterator* dict_it = ds_stream_index_get_dictionaries(stream);
+	if (oscap_string_iterator_has_more(dict_it)) {
+		printf("Dictionaries:\n");
+	} else {
+		printf("No dictionaries.\n");
+	}
+	while (oscap_string_iterator_has_more(dict_it)) {
+		const char * id = oscap_string_iterator_next(dict_it);
+		printf("\tRef-Id: %s\n", id);
+	}
+	oscap_string_iterator_free(dict_it);
+	return OSCAP_OK;
+}
+
+const char *benchmark_get_profile_or_report_bad_id(struct xccdf_benchmark *bench, const char *profile_suffix, const char *source_file)
+{
+	oscap_profile_match_t match_status;
+	const char *result = xccdf_benchmark_match_profile_id(bench, profile_suffix, &match_status);
+	evaluate_suffix_match_result(match_status, profile_suffix, source_file);
+	return result;
+}
+
+static void app_info_single_benchmark(struct xccdf_benchmark *bench, const struct oscap_action *action, struct oscap_source *source)
+{
+	if (action->show_profiles_only) {
+		_print_single_benchmark_profiles_only(bench);
+		xccdf_benchmark_free(bench);
+	} else if (action->profile) {
+		const char *profile_id = benchmark_get_profile_or_report_bad_id(bench, action->profile, action->file);
+		if (profile_id != NULL) {
```

Comment:
Why the suffix match isn't used for datastreams as well? With XCCDF it's nice, but not so much useful. The benefit of suffix match is when the ID is long, which happens in datastreams.

The behavior of "oscap info --profile ID DS" should be same as "oscap info --profile ID XCCDF". Please use this function with datastreams as well.

Compare:
```
[jcerny@thinkpad openscap{feature/info-improvement}]$ myoscap info --profile sss /usr/share/xml/scap/ssg/content/ssg-rhel7-ds.xml                                                                                   
Document type: Source Data Stream                    
Imported: 2017-08-29T22:29:14                        

Stream: scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml                                  
Generated: (null)         
Version: 1.2              
[jcerny@thinkpad openscap{feature/info-improvement}]$ myoscap info --profile sss /usr/share/xml/scap/ssg/content/ssg-rhel7-xccdf.xml                                                                                
Document type: XCCDF Checklist                       
No profile matching suffix "sss" was found. Get available profiles using:                                 
$ oscap info "/usr/share/xml/scap/ssg/content/ssg-rhel7-xccdf.xml"                                        
[jcerny@thinkpad openscap{feature/info-improvement}]$ 
```
The first command output is inappropriate, it should have look as the second.

Reply:
oops, I forgot about that.